pkg/kubelet/kubeletconfig/util/codec/codec.go

```diff
@@ -26,9 +26,37 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/scheme"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/v1beta1"
 )
 
-// TODO(mtaufen): allow an encoder to be injected into checkpoint objects at creation time? (then we could ultimately instantiate only one encoder)
+// EncodeKubeletConfig encodes an internal KubeletConfiguration to an external JSON representation
+func EncodeKubeletConfig(internal *kubeletconfig.KubeletConfiguration) ([]byte, error) {
+	encoder, err := NewKubeletConfigJSONEncoder()
+	if err != nil {
+		return nil, err
+	}
+	// encoder will convert to external version
+	data, err := runtime.Encode(encoder, internal)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+// NewKubeletConfigJSONEncoder returns an encoder that can write a KubeletConfig to JSON
+func NewKubeletConfigJSONEncoder() (runtime.Encoder, error) {
```

Comment:
I would rather not export this if it is only used by `EncodeKubeletConfig`.  We can always make it an exported method at a later date.

Reply:
I think I had planned on refactoring the e2e node tests to use this, instead of the duplicate in util.go. I will add that to this PR.

Reply:
Updated.